code/game/objects/items/RPD.dm

```diff
@@ -366,37 +415,49 @@ GLOBAL_LIST_INIT(fluid_duct_recipes, list(
 	if(!user.IsAdvancedToolUser() || istype(A, /turf/open/space/transit))
 		return ..()
 
+	var/atom/attack_target = A
+
 	//So that changing the menu settings doesn't affect the pipes already being built.
 	var/queued_p_type = recipe.id
 	var/queued_p_dir = p_dir
 	var/queued_p_flipped = p_flipped
 
+	//Unwrench pipe before we build one over/paint it, but only if we're not already running a do_after on it already to prevent a potential runtime.
+	if((mode & DESTROY_MODE) && (upgrade_flags & RPD_UPGRADE_UNWRENCH) && istype(attack_target, /obj/machinery/atmospherics) && !(attack_target in user.do_afters))
+		attack_target = attack_target.wrench_act(user, src)
+		if(attack_target == TRUE)
+			return ..()
+		if(!isatom(attack_target))
+			CRASH("When attempting to call [A.type].wrench_act(), received the following non-atom return value: [attack_target]")
+
 	//make sure what we're clicking is valid for the current category
 	var/static/list/make_pipe_whitelist
 	if(!make_pipe_whitelist)
 		make_pipe_whitelist = typecacheof(list(/obj/structure/lattice, /obj/structure/girder, /obj/item/pipe, /obj/structure/window, /obj/structure/grille))
+	if(istype(attack_target, /obj/machinery/atmospherics) && ((mode & BUILD_MODE) && !(mode & PAINT_MODE))) //Reduces pixelhunt when coloring is off.
```

Comment:
good change, comment could be clearer though (target turf if on buildmode so that it doesn't try painting a pipe you click on)